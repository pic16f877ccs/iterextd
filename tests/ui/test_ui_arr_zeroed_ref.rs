use iterextd::IterExtd;

fn main() {
    let mut arr = [10u8, 11, 22, 33, 44, 55, 66, 77, 88, 99, 111, 222];
    let iter = arr.iter().arr_chunks::<5>();
    let _: (usize, [_; 2]) = iter.collect_arr_zeroed();
}
