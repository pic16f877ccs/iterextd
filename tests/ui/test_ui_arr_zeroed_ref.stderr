error[E0277]: the trait bound `&u8: iterextd::iterator::AllowZero` is not satisfied
 --> tests/ui/test_ui_arr_zeroed_ref.rs:6:35
  |
6 |     let _: (usize, [_; 2]) = iter.collect_arr_zeroed();
  |                                   ^^^^^^^^^^^^^^^^^^ the trait `iterextd::iterator::AllowZero` is not implemented for `&u8`
  |
  = help: the trait `iterextd::iterator::AllowZero` is implemented for `u8`
  = note: required for `[&u8; 5]` to implement `iterextd::iterator::AllowZero`
note: required by a bound in `collect_arr_zeroed`
 --> src/iterator.rs
  |
  |     fn collect_arr_zeroed<const N: usize>(self) -> (usize, [Self::Item; N])
  |        ------------------ required by a bound in this associated function
...
  |         Self::Item: AllowZero,
  |                     ^^^^^^^^^ required by this bound in `IterExtd::collect_arr_zeroed`
